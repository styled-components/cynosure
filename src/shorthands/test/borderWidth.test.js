// @flow
import borderWidth from '../borderWidth'

describe('borderWidth', () => {
  it('properly applies a value when passed only one', () => {
    expect(borderWidth('12px')).toEqual({
      borderBottomWidth: '12px',
      borderLeftWidth: '12px',
      borderRightWidth: '12px',
      borderTopWidth: '12px',
    })
  })
  it('properly applies values when passed two', () => {
    expect(borderWidth('12px', '24px')).toEqual({
      borderBottomWidth: '12px',
      borderLeftWidth: '24px',
      borderRightWidth: '24px',
      borderTopWidth: '12px',
    })
  })
  it('properly applies values when passed three', () => {
    expect(borderWidth('12px', '24px', '36px')).toEqual({
      borderBottomWidth: '36px',
      borderLeftWidth: '24px',
      borderRightWidth: '24px',
      borderTopWidth: '12px',
    })
  })
  it('properly applies values when passed four', () => {
    expect(borderWidth('12px', '24px', '36px', '48px')).toEqual({
      borderBottomWidth: '36px',
      borderLeftWidth: '48px',
      borderRightWidth: '24px',
      borderTopWidth: '12px',
    })
  })
  it('properly applies values when passed integers', () => {
    expect(borderWidth(12, 24, 36, 48)).toEqual({
      borderBottomWidth: 36,
      borderLeftWidth: 48,
      borderRightWidth: 24,
      borderTopWidth: 12,
    })
  })
  it('properly applies values when passed zero', () => {
    expect(borderWidth(0)).toEqual({
      borderBottomWidth: 0,
      borderLeftWidth: 0,
      borderRightWidth: 0,
      borderTopWidth: 0,
    })
  })
})
