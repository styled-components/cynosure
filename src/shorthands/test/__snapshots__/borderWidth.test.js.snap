exports[`borderWidth properly applies a value when passed only one 1`] = `
Object {
  "borderBottomWidth": "12px",
  "borderLeftWidth": "12px",
  "borderRightWidth": "12px",
  "borderTopWidth": "12px",
}
`;

exports[`borderWidth properly applies values when passed four 1`] = `
Object {
  "borderBottomWidth": "36px",
  "borderLeftWidth": "48px",
  "borderRightWidth": "24px",
  "borderTopWidth": "12px",
}
`;

exports[`borderWidth properly applies values when passed three 1`] = `
Object {
  "borderBottomWidth": "36px",
  "borderLeftWidth": "24px",
  "borderRightWidth": "24px",
  "borderTopWidth": "12px",
}
`;

exports[`borderWidth properly applies values when passed two 1`] = `
Object {
  "borderBottomWidth": "12px",
  "borderLeftWidth": "24px",
  "borderRightWidth": "24px",
  "borderTopWidth": "12px",
}
`;
